website/projects/awssync.py

```diff
@@ -22,8 +26,37 @@ def button_pressed(self):
         :return: True if function executes successfully
         """
         self.logger.info("Pressed button")
+        self.logger.info(self.get_emails_with_teamids())
         return True
 
+    def get_all_mailing_lists(self):
+        """
+        Get all mailing lists from the database.
+
+        :return: List of mailing lists
+        """
+        mailing_lists = MailingList.objects.all()
+        mailing_list_names = [ml.email_address for ml in mailing_lists]
+        return mailing_list_names
+
+    def get_emails_with_teamids(self):
+        """
+        Create a tuple with email and corresponding teamID, where teamID is a concatenation of ID and semesterID.
+
+        :return: list of (email, teamid)
+        """
+        mailing_lists = MailingList.objects.all()
+        email_id = []
+        for ml in mailing_lists:
+            project = ml.projects.all()
+            if len(project) != 0:
+                project_id = [(p.id, p.semester.id) for p in project]
+                project_id = int(str(project_id[0][0]) + str(project_id[0][1]))
```

Comment:
You are concatenating two primary keys from different fields with the goal of creating a unique identifier of type integer. However, the resulting identifier can be reconstructed in more than one way, making the mapping non-injective. This will lead to bugs.

Example: 
`int(str(133) + str(7)) == int(str(1) + str(337))`

Reply:
This is indeed a concern, ill will look at in combination with the comment about the unique_together attribute and resolve it

Reply:
The new way you've done it is still non-injective, but the chance of it occurring is neglegible compared to earlier.